linalg/hypre.hpp

```diff
@@ -229,15 +233,22 @@ class HypreParMatrix : public Operator
 
    /// Converts hypre's format to HypreParMatrix
    /** If @a owner is false, ownership of @a a is not transferred */
-   explicit HypreParMatrix(hypre_ParCSRMatrix *a, bool owner = true)
+   void WrapHypreParCSRMatrix(hypre_ParCSRMatrix *a, bool owner = true)
    {
-      Init();
       A = a;
       if (!owner) { ParCSROwner = 0; }
       height = GetNumRows();
       width = GetNumCols();
    }
```

Comment:
The new method `WrapHypreParCSRMatrix` has to destroy the old contents of the `HypreParMatrix` first, e.g. by calling the private method `Destroy()`. Then it has to reset/re-initialize all data members, either by calling the private method `Init()` or by re-initializing in-place the rest of the data members.